Code Review

TheQuestCrafter opened the pull request.

TheQuestCrafter committed:
Re-adding scripts for code review

dantogno commented on Assets/Scripts/CameraController.cs:
```diff
@@ -0,0 +1,18 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour {
```
Put all the curly brackets on a new line

dantogno replied:
Are you using Cinemachine? I'd like you to use cinemachine. And then just delete this script because I assume it will be unused at that point.

dantogno commented on Assets/Scripts/CameraController.cs:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    // Update is called once per frame
+    void LateUpdate () {
```
{ on new line

dantogno commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,95 @@
+    private Sprite inactiveSprite, activeSprite;
+
+    [SerializeField]
+    private Color unactivatedColor, activatedColor;
```
I think "inactive" is probably more grammatically correct 🤔. Also more consistent with `inactiveScale` variables you have.

dantogno commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,95 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour {
```
{ on new line

dantogno commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,95 @@
+    {
+        //checkpoint changes sprite to open book when updated
+        isActivated = value;
+        //UpdateScale();
```
Remove commented out code

dantogno replied:
If you aren't using those functions, remove them

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Collectible : MonoBehaviour {
```
{ on new line

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+public class Collectible : MonoBehaviour {
+
+    [SerializeField]
+    private int upDownAmount=10, timeTextIsUp;
```
Declare these individually and use ToolTip attributes to explain to me what they are for. Also, I'm guessing they may make sense as floats but I'll decide once I understand what they are better. https://docs.google.com/document/d/1yrKCY_sVAdWkwgWAjJLiJFj1OOHn6dC__pGkSVB61YY/edit#heading=h.35gp8tzfruuq

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    private CircleCollider2D circleCollider;
+    private bool textIsActive, textWasActive;
+    private Color fillerColor;
+    //private PlayerCharacter PC;
```
remove commented out code

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    private Text collectibleText;
+
+    private int bounceAmount, UITime=0;
+    private bool falling=true;
```
add spaces before and after operators  e.g. `falling = true;`

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    [SerializeField]
+    private Text collectibleText;
+
+    private int bounceAmount, UITime=0;
```
time and amounts should almost never be ints. They should be floats. Unless there's something I'm not following here.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    private string objectName;
+    [SerializeField]
+    private float bounceYAmount=0.005f;
+    [SerializeField]
```
This is probably enough serialized fields to use a #region https://docs.google.com/document/d/1yrKCY_sVAdWkwgWAjJLiJFj1OOHn6dC__pGkSVB61YY/edit#heading=h.oezlj2pye3ju

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+            collectibleText.gameObject.SetActive(true);
+            Debug.Log(UITime + " left");
+            textIsActive = true;
+            UITime--;
```
This is deceptive because time indicates a standard unit of time like a second, but you are incrementing an int every fixed update, so the time is based on fixed update interval, which is at least consistent but I doubt you are sure what it is, and we don't want uncertainty. 

Use a float and increment it by Time.DeltaTime to get a better timer functionality. https://unity3d.com/learn/tutorials/topics/scripting/delta-time

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+            textIsActive = true;
+            UITime--;
+            fillerColor = collectibleText.color;
+            fillerColor.a -= 0.001f;
```
replace magic number with named variable.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    }
+
+    /*public void Collected()
+    {
```
remove commented out code

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    }*/
+
+    //Allows player to pick up collectible
+    //ToDo: Make collectible display text in UI
```
Is this done?

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+        audioSource.Play();
+    }*/
+
+    //Allows player to pick up collectible
```
Replace this comment with a summary (hotkey: type /// in visual studio) https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/summary

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            //UITime = PC.CollectibleTimerReset(UITime, timeTextIsUp);
```
remove commented out code

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+        if (other.gameObject.CompareTag("Player"))
+        {
+            //UITime = PC.CollectibleTimerReset(UITime, timeTextIsUp);
+            if (objectName == "Necklace")
```
"necklace" is a magic string. Replace with a named variable. If it never changes, it can be marked as a constant ( `const string` )

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+            //UITime = PC.CollectibleTimerReset(UITime, timeTextIsUp);
+            if (objectName == "Necklace")
+            {
+                Debug.Log("Picked Up Necklace");
```
Remove debug code. Surprisingly, in Unity, Debug.Log actually effects performance in release builds! Pretty dumb.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+                ResetAlpha();
+                UITime = timeTextIsUp;
+            }
+            else if (objectName == "Present")
```
magic string

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+            audioSource.Play();
+            spriteRenderer.enabled = false;
+            circleCollider.enabled = false;
+            //Destroy(gameObject, audioSource.clip.length);
```
remove commented out code

dantogno replied:
Do you want to destroy this after the time text is up elapses? If you want to destroy it, we can figure out the right time.

dantogno replied:
But honestly it's prob not necessary.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+        collectibleText.color = fillerColor;
+    }
+
+    //Makes the collectible bounce down and then back up to its starting position.
```
Replace with summary https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/summary

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+    }
+
+    //Makes the collectible bounce down and then back up to its starting position.
+    //ToDo: add lerping
```
https://docs.unity3d.com/ScriptReference/Mathf.Lerp.html

TheQuestCrafter replied:
With my current workload, I won't be able to implement this feature to replace the current which already works for how I want it. Will look into it for the future though.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+        if (bounceAmount <= upDownAmount && falling && bounceAmount>0)
+        {
+            Debug.Log("Falling");
+            GameObject.Find(objectName).transform.position = new Vector2(GameObject.Find(objectName).transform.position.x, GameObject.Find(objectName).transform.position.y - bounceYAmount);
```
You don't want to call GameObject.Find in update or fixed update. It's expensive and almost always unnecessary to call more than once. Instead, call it once in `Start` or `Awake` and "cache" the result (i.e. store in a variable). 

But honestly, I question whether any GameObject.Find call is necessary. What are you searching for? Typically I'd avoid these calls period because there is a chance the names can get messed up / changed either in code or in the Unity editor, and then your game stops working. Remove chances for human error.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+        }
+        else if (falling == false && bounceAmount < upDownAmount)
+        {
+            Debug.Log("Rising");
```
Remove debug code

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -0,0 +1,134 @@
+            Debug.Log("Falling=false");
+            falling = false;
+        }
+        else if(falling==false && bounceAmount >= upDownAmount)
```
add spaces surrounding operators `falling == false`.

dantogno replied:
You normally don't want to use `== false` or `true` for bools. You can just say `if (!falling)` or `if (falling)`

dantogno commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,43 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Door : MonoBehaviour {
```
new line {

dantogno commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,43 @@
+            //ToDo: Replace scene with variable so each door can have a different scene to lead to.
+            //audioSource.Play();
+            Debug.Log("Player Activated Door");
+            SceneManager.LoadScene("SampleScene");
```
You don't want to hard code your door

dantogno commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,43 @@
+        {
+            //allows player to exit current scene and move to next scene
+            //ToDo: Replace scene with variable so each door can have a different scene to lead to.
+            //audioSource.Play();
```
Sounds not working here because you are loading new scene before sound is done. You'd have to either tell an audio source that is on a game object marked [DontDestroyOnLoad ](https://docs.unity3d.com/ScriptReference/Object.DontDestroyOnLoad.html) to play or not load until the sound is done.

dantogno commented on Assets/Scripts/Hazard.cs:
```diff
@@ -0,0 +1,23 @@
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.CompareTag("Player"))
```
Extract all of this into it's own method with a good name and call the method from `OnTriggerEnter`

dantogno commented on Assets/Scripts/Hazard.cs:
```diff
@@ -0,0 +1,23 @@
+    {
+        if (collision.gameObject.CompareTag("Player"))
+        {
+            audioSource.PlayDelayed(0.5f);
```
magic number (`0.5f`)

dantogno commented on Assets/Scripts/Hazard.cs:
```diff
@@ -0,0 +1,23 @@
+            audioSource.PlayDelayed(0.5f);
+            PlayerCharacter player = collision.GetComponent<PlayerCharacter>();
+            player.StartRespawn();
+            //plays death sound and starts the player's respawn.
```
Move comment to summary tag for the new method I told you to make.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+public class PlayerCharacter : MonoBehaviour {
+
+    //public int UITextTime { get; set; }
```
remove commented out code

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class PlayerCharacter : MonoBehaviour {
```
{ new line

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+public class PlayerCharacter : MonoBehaviour {
+
+    //public int UITextTime { get; set; }
+
```
remove white space

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    //public int UITextTime { get; set; }
+
+    [SerializeField]
```
Wrap all these serialized fields in a `#region` labeled "Serialized fields"

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private Rigidbody2D myRigidBody;
+    [SerializeField]
+    private float acceleration;
+    private float direction;
```
Move all the plain old private fields (not serialized) together into their own `#region` called "Private fields".

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    [SerializeField]
+    private float jumpCooldown;
+    [SerializeField]
+    private float jumpOneModifier;
```
add `ToolTip` attribute https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private float jumpCooldown;
+    [SerializeField]
+    private float jumpOneModifier;
+    [SerializeField]
```
add `ToolTip` attribute https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private float jumpOneModifier;
+    [SerializeField]
+    private float jumpTwoModifier;
+    [SerializeField]
```
add `ToolTip` attribute https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    [SerializeField]
+    LayerMask whatIsGround;
+    [SerializeField]
+    float groundRadius = 0.2f, speed;
```
explicitly write out `private` before all private fields.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    [SerializeField]
+    float groundRadius = 0.2f, speed;
+    bool facingRight = true, grounded = false, falling=false;
+    private bool isInDeath;
```
name idea: `isDead`?

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    bool facingRight = true, grounded = false, falling=false;
+    private bool isInDeath;
+    [SerializeField]
+    private float timeToDeath;
```
add `ToolTip` attribute https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private AudioSource audioSource;
+    private AudioSource collectibleAudio;
+    [SerializeField]
+    private float deathVector;
```
add `ToolTip` attribute https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    [SerializeField]
+    private float deathVector;
+
+
```
remove extra white space. one blank line separating fields from methods is good.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private float deathVector;
+
+
+    // Use this for initialization
```
Remove comment

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        audioSource = GetComponent<AudioSource>();
+        jumpNumber = 0;
+        jumpCooldown = 0;
+        anim.SetBool("IsDead", false); //starting game not dead
```
you have another bool about being dead. Use this instead of hardcoding false?

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    // Update is called once per frame
+    void Update() {
+        if (Time.realtimeSinceStartup - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
```
I have a feeling this can be refactored / simplified.  https://unity3d.com/learn/tutorials/topics/scripting/delta-time This is on canvas for XP if you haven't done it.

dantogno replied:
In advanced game engine scripting I'll go over coroutines, which are easier for timers IMO

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        {
+            if (Input.GetButtonDown("Jump") && jumpCooldown == 0)
+            {
+                Jump();//allows player to jump and checks to make sure character can't jump when dead
```
Put this comment in a summary for the Jump function.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+            anim.SetBool("IsDead", true);
+            if (Time.realtimeSinceStartup - deathTimer >= timeToDeath)
+            {
+                Respawn();//if death timer has gone up and matches time to death, it will enter the next stage of respawn.
```
perhaps extract this if statement into it's own method and add this comment as a summary.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+            if (Time.realtimeSinceStartup - deathTimer >= timeToDeath)
+            {
+                Respawn();//if death timer has gone up and matches time to death, it will enter the next stage of respawn.
+                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
```
You should have a cached reference to the player's rigidbody2D that you can reuse rather than calling `GetComponent` multiple times. It's an expensive call, so you want to call it once in `Start` or `Awake` and "cache" the result in a variable.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+                Respawn();//if death timer has gone up and matches time to death, it will enter the next stage of respawn.
+                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
+            }
+            GetComponent<Rigidbody2D>().velocity = myRigidBody.velocity/deathVector;
```
Make this it's own method with a good name / summary so I understand what it's doing / why.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        GetMovementInput();
+        if (jumpNumber > 0)
+        {
+            speedJumpReducer = 2; // reduces jump height with each subsequent jump.
```
Extract all this jump stuff into it's own method with good name / summary.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        {
+            speedJumpReducer = 2; // reduces jump height with each subsequent jump.
+        }
+        /*if (Input.GetKey(KeyCode.Space) && jumpCooldown == 0)
```
remove commented out code

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        grounded = Physics2D.OverlapCircle(groundCheck.position, groundRadius, whatIsGround);
+        anim.SetBool("IsOnGround", grounded);
+        //sets tells animator if character is falling or on the ground.
+        if (myRigidBody.velocity.y < 0 && grounded == false)
```
Wrap the falling logic into it's own method `UpdateIsFalling` (I'd rename falling to `isFalling`) and call it in `FixedUpdate`. Leave the anim parameter part here though. Just update isFalling in UpdateIsFalling. Set the falling anim param in the update anim params method.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    private void UpdateAnimationParameters()
+    {
+        grounded = Physics2D.OverlapCircle(groundCheck.position, groundRadius, whatIsGround);
```
move this to an `UpdateIsOnGround` method called in `FixedUpdate`. (I'd rename `grounded` to "isOnGround")

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private void FixedUpdate()
+    {
+        UpdatePhysicsMaterial();
+        if (Time.time - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
```
magic number.

dantogno replied:
Also, you're going to want to just have one variable for `isDead` and check it whenever you need to know if you're alive or dead instead of doing all this confusing conditional logic over and over.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        if (Time.time - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        {
+            //character cannot use the move function when dying.
+            if (Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.A))
```
Don't use `GetKey` ever. Use `GetButton` or `GetAxis` and the InputManager https://docs.unity3d.com/Manual/class-InputManager.html

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+            {
+                Move();//lets player move a fixed amount not dependent on frames
+            }
+            //player flips sprite when going opposite direction of prior movement.
```
You could probably extract this into a `FlipIfNeeded` method (or pick your own name that makes sense) with a summary.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+            {
+                Flip();
+            }
+            deathTimer = 0;
```
Move all the death timer stuff to it's own method.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    private void GetMovementInput()
+    {
+        //sets direction of movement from player commands
```
Move comment to summary tag.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        Vector2 clampedVelocity = myRigidBody.velocity;
+        clampedVelocity.x = Mathf.Clamp(myRigidBody.velocity.x, -maxSpeed / speedJumpReducer, maxSpeed / speedJumpReducer);
+        /*if (isInDeath)
+        {
```
Remove commented out code

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private void GetMovementInput()
+    {
+        //sets direction of movement from player commands
+        direction = Input.GetAxisRaw("Horizontal");
```
Technically this is a bit more than direction unless you are rounding it to 1 or negative 1 every time. It's direction and magnitude (e.g. if you were using analog stick, you could walk more slowly by not fully pushing the stick in a direction)

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    {
+        myRigidBody.AddForce(Vector2.right * direction * acceleration / speedJumpReducer);
+        Vector2 clampedVelocity = myRigidBody.velocity;
+        clampedVelocity.x = Mathf.Clamp(myRigidBody.velocity.x, -maxSpeed / speedJumpReducer, maxSpeed / speedJumpReducer);
```
Explain jumpspeed reducer in summary tag

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        {
+            clampedVelocity.x = clampedVelocity.x/10;
+        }*/
+        myRigidBody.velocity = clampedVelocity;//adds up a build up of acceleration to a maximum velocity
```
I think this is straightforward enough to ditch the comment, or move it to summary.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    private void Jump()
+    {
+        //Allows player to jump twice before having to fall to the ground.
```
move to summary

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private void Jump()
+    {
+        //Allows player to jump twice before having to fall to the ground.
+       if (jumpNumber == 0&&grounded==true)
```
add spaces between values and operators, don't use `== true` just state the name of the bool (it resolves to true or false) (e.g. `jumpNumber == 0 && grounded)

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+            jumpCooldown = 90;
+            jumpNumber++;
+        }
+        
```
remove whitespace

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    public void StartRespawn()
+    {
+        //stage one of respawn
```
make summary explaining why this is a method you need

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    private void Respawn()
+    {
+        //last stage of respawn, checks for checkpoint and resets 'is dead' to default of false
```
move to summary

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    private void Respawn()
+    {
+        //last stage of respawn, checks for checkpoint and resets 'is dead' to default of false
+        anim.SetBool("IsDead", false);
```
move to your `UpateAnimationParameters` function.

dantogno replied:
Don't hardcode it, have an `isDead` bool that is up to date and set the anim param to `isDead` in your `UpdateAnimParameters`

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+        else
+        {
+            SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+        }
```
fix indents

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+
+    public void SetCurrentCheckpoint(Checkpoint newCurrentCheckpoint)
+    {
+        //creates checkpoint for player.
```
move to summary

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    }
+    void Flip()
+    {
+        //flips player when switching direction in which the player is facing.
```
move to summary

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+            currentCheckpoint = newCurrentCheckpoint;
+            currentCheckpoint.SetIsActivated(true);
+    }
+    void Flip()
```
explicitly specify whether each method is private or public, etc.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -0,0 +1,256 @@
+    }
+
+    /*public int CollectibleTimerReset(int UITimer, int UIMaxTimer)
+    {
```
remove commented out code

dantogno reviewed (changes requested):
Nice work @TheQuestCrafter! I left you quite a bit of action items here, but that's just how you learn (plus most of them are just me enforcing style rules that not everyone agrees on--though the official C# language bosses do).

Please make commits that address all of this, or leave replies to my comments here if you have questions or concerns about any of them. Time is a bit of the essence, because I'd love to have you review another student!

TheQuestCrafter committed:
Fixed Cinemachine and Removed Camera Controller

TheQuestCrafter committed:
Quick Fixes from Code Review

Including removing commented code, removing whitespace, fixing variable names,  privating some variables, removing extraneous methods,  fixed indentations, and removed removed debugs.

TheQuestCrafter committed:
Edited Cinemachine's View

TheQuestCrafter committed:
Main Menu added

Followed Unity tutorial for adding a main menu for the game.

TheQuestCrafter committed:
Level 1 and Customized Main Menu

Changed and edited Sample Scene into a level 1. Edited main menu to customize it to the current game. Changed Door script so it used a serialized field.

TheQuestCrafter committed:
Main Menu Update

TheQuestCrafter committed:
Edited Level 1 and Added Level 2

TheQuestCrafter committed:
Edited Levels 1 & 2 and added level 3 with some minor fixes

Also added silent and invisible checkpoint so that when the player is killed after recieving a collectible, it does not reappear.

dantogno commented:
@TheQuestCrafter done addressing these items?

TheQuestCrafter commented:
Not quite yet, I’ve been swamped this week with a couple of early finals. I plan on getting to them today when I get some free time after a class. I will email you/message you on discord when I’ve finished. Sorry about the wait.

> On Dec 6, 2018, at 11:11 AM, David Antognoli <notifications@github.com> wrote:
> 
> @TheQuestCrafter done addressing these items?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

TheQuestCrafter committed:
Summaries, Tooltips, and Refinements.

TheQuestCrafter committed:
Better Formatting of Code

TheQuestCrafter committed:
Door Sound and Final Updates for Code Review

TheQuestCrafter commented:
I am now finished going through the changes requested.

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -18,14 +22,19 @@ public class Collectible : MonoBehaviour
     private CircleCollider2D circleCollider;
     private bool textIsActive, textWasActive;
     private Color fillerColor;
+    const string necklace = "Necklace", gem = "Gem", ring = "WeddingRing", present = "Present";
```
there's a C# trick for this, although I don't think it works in Unity by default until 2018.3 so don't worry about changing it for now.  https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/nameof

dantogno replied:
e.g. `const string necklace = nameof(necklace);` then there are no human typed strings at all so no possibility of typos

dantogno commented on Assets/Scripts/Collectible.cs:
```diff
@@ -105,19 +120,19 @@ private void ResetAlpha()
     //ToDo: add lerping
     private void Bounce()
     {
+        //<summary>Makes the collectible bounce down and then back up to its starting position.</summary>
```
summaries go just outside (above) the name of the method / class.

dantogno commented on Assets/Scripts/Door.cs:
```diff
@@ -35,9 +35,10 @@ private void OnTriggerExit2D(Collider2D collision)
 	void Update () {
         if (Input.GetButtonDown("Activate") && isPlayerInTrigger)
         {
+            DontDestroyOnLoad(this.gameObject);
```
This will leave your door around. it's nice for making sure the sound plays, but now you have a door in the next level that you might not want...

TheQuestCrafter replied:
Is there a way to just delay the destroy on load?

dantogno commented on Assets/Scripts/Hazard.cs:
```diff
@@ -1,21 +1,29 @@
 
     private void OnTriggerEnter2D(Collider2D collision)
+    {
+        StartDeath(collision);
```
I know "collision" is the template parameter name, but do you think it reads more clearly if it were named something like `objectCollidedWith`?

dantogno replied:
I guess it's a `Collider`, so maybe it should be `otherCollider`. Just thoughts.

dantogno commented on Assets/Scripts/Hazard.cs:
```diff
@@ -1,21 +1,29 @@
+            audioSource.PlayDelayed(audioDelay);
             PlayerCharacter player = collision.GetComponent<PlayerCharacter>();
             player.StartRespawn();
             //plays death sound and starts the player's respawn.
```
I'd convert this comment to a summary tag for `StartDeath`

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -35,17 +38,25 @@ public class PlayerCharacter : MonoBehaviour
-    private bool isInDeath;
     [SerializeField]
+    [Tooltip("The vector multiplier that slows the character to halt while dying.")]
+    private float deathVector;
```
So does it need to be greater than 0 but less than 1? You could use the Range attribute, or at least clarify the tooltip. https://docs.unity3d.com/ScriptReference/RangeAttribute.html

If we can stop the designer from putting in a dumb value, we should.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -57,87 +68,127 @@ public class PlayerCharacter : MonoBehaviour
-                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
             }
-            GetComponent<Rigidbody2D>().velocity = myRigidBody.velocity/deathVector;
+            DeathTimerZero();
```
I'd prefer method names to be verbs. Tell me what action this takes in its name.

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -57,87 +68,127 @@ public class PlayerCharacter : MonoBehaviour
+
+    private void JumpCheck()
+    {
+        ///Checks what jump the character is on and applies and checks cooldown as well as apply x movement reduction while jumping
```
move this to a summary tag the line above the method declaration

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -57,87 +68,127 @@ public class PlayerCharacter : MonoBehaviour
+
+    private void DyingUpdate()
+    {
+        //<summary>if death timer has gone up and matches time to death, it will enter the next stage of respawn.</summary>
```
summaries go above the method declaration

dantogno commented on Assets/Scripts/PlayerCharacter.cs:
```diff
@@ -57,87 +68,127 @@ public class PlayerCharacter : MonoBehaviour
-        anim.SetBool("IsOnGround", grounded);
-        //sets tells animator if character is falling or on the ground.
-        if (myRigidBody.velocity.y < 0 && grounded == false)
+        IsOnGround();
```
Make this method name a verb. e.g. `UpdateIsOnGround`. Name it after what it does.

TheQuestCrafter committed:
Summary Edit

TheQuestCrafter committed:
Spike Art and Door Art

TheQuestCrafter committed:
Itch.io Readme
